R/R/install.R

```diff
@@ -0,0 +1,154 @@
+#' Install Metaflow Python package
+#'
+#' This function wraps installation functions from [reticulate][reticulate::reticulate] to install the Python packages
+#' **metaflow** and it's Python dependencies.
+#'
+#' This package uses the [reticulate][reticulate::reticulate] package
+#' to make an interface with the [Metaflow](https://metaflow.org/)
+#' Python package.
+#'
+#' @param method `character`, indicates to use `"conda"` or `"virtualenv"`.
+#' @param prompt boolean, whether or not to prompt user for confirmation before installation. Default is TRUE.
+#' @param version `character`, version of Metaflow to install. The default version
+#' is the latest available on PyPi.
+#' @param ... other arguments sent to [reticulate::conda_install()] or
+#'    [reticulate::virtualenv_install()]
+#'
+#' @seealso
+#' [reticulate: Using reticulate in an R Package](https://rstudio.github.io/reticulate/articles/package.html),
+#' [reticulate: Installing Python Packages](https://rstudio.github.io/reticulate/articles/python_packages.html)
+#' @examples
+#' \dontrun{
+#' # not run because it requires Python
+#' install_metaflow()
+#' }
+#' @export
+install_metaflow <- function(method = c("conda", "virtualenv"),
+                             prompt = TRUE,
+                             version = NULL,
+                             ...) {
+  envname <- pkg.env$envname
+
+  env_set <- check_environment(envname)
+  if (method == "conda" && env_set[["virtualenv"]]) {
+    message("A virtualenv environment is detected for Metaflow.")
+    message("Can't install within a virtualenv using conda.")
+    stop("Please remove existing virtualenv by `remove_metaflow()` and run the install again.")
+  }
+
+  if (method == "virtualenv" && env_set[["conda"]]) {
+    message("A conda environment is detected for Metaflow.")
+    message("Can't install within a conda environment using virtualenv.")
+    stop("Please remove existing conda env by `remove_metaflow()` and run the install again.")
+  }
+
+  # validate stage, method arguments
+  method <- match.arg(method)
+
+  # conda and pip use different syntax for indicating versions
+  if (identical(method, "conda")) {
+    version_sep <- "="
+  } else {
+    version_sep <- "=="
+  }
+
+  if (is.null(version)) {
+    metaflow_pkg_version <- "metaflow"
+  } else {
+    metaflow_pkg_version <- paste("metaflow", version, sep = version_sep)
+  }
+
+  packages <- c(metaflow_pkg_version, "numpy", "pandas")
+
+  # create environment if not present
+  if (method == "conda") {
+    conda <- tryCatch(reticulate::conda_binary(),
+      error = function(e) NULL
+    )
+    have_conda <- !is.null(conda)
+    if (!have_conda) {
+      message("No conda was found in the system.")
+      message("Miniconda is an open source environment management system for Python.")
```

Comment:
For the sake of correctness - `Miniconda is an open-source package management and environment management system.` Conda is not limited to Python.